src/chainstate/stacks/events.rs

```diff
@@ -63,6 +65,128 @@ pub struct StacksTransactionReceipt {
     pub microblock_header: Option<StacksMicroblockHeader>,
 }
 
+/// Represents a successful transaction. This transaction should be added to the block.
+#[derive(Debug, Clone, PartialEq)]
+pub struct TransactionSuccess {
+    pub tx: StacksTransaction,
+    // The fee that was charged to the user for doing this transaction.
+    pub fee: u64,
+    pub receipt: StacksTransactionReceipt,
+}
+
+/// Represents a failed transaction. Something concreteley went wrong.
+#[derive(Debug)]
+pub struct TransactionError {
+    pub tx: StacksTransaction,
+    // Note: This should be an `Error` when checked in.
+    pub error: Error,
+}
+
+/// Represents a transaction that was skipped, but might succeed later.
+#[derive(Debug, Clone, PartialEq)]
+pub struct TransactionSkipped {
+    pub tx: StacksTransaction,
+    pub reason: String,
+}
+
+/// `TransactionResult` represents the outcome of transaction processing.
```

Comment:
Really, this is a `MiningResult`.  A transaction cannot be skipped by the blockchain-processing logic.  That the miner uses the blockchain-processing logic to produce a block does not mean that the blockchain-processing logic needs to be aware of the decision to include or exclude the transaction when it is mined.